File: Codeforce/2023/RPC12/Idle.py
n,k=map(int,input().split())
procesos = [int(x) for x in input().split()]
EnProcesamiento=[]
maximo = max(procesos) if k == 1 else -1
while len(procesos)>0 or len(EnProcesamiento)>0:
    for i in range(k-len(EnProcesamiento)):
        if len(procesos)>0:
            EnProcesamiento.append(procesos.pop(0))
    value = min(EnProcesamiento)
    EnProcesamiento=[elemento-value for elemento in EnProcesamiento if elemento-value>0]
    maximo=max(maximo,value)
print(maximo)

File: Codeforce/2023/RPC12/Idle.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <deque>
using namespace std;

int main()
{
    int n, k;
    cin >> n >> k;

    deque<int> procesos;
    for (int i = 0; i < n; i++)
    {
        int monda;
        cin >> monda;
        procesos.push_back(monda);
    }

    vector<int> en_procesamiento;
    int maximo;

    if (k == 1)
    {
        maximo = *max_element(procesos.begin(), procesos.end());
    }
    else
    {
        maximo = -1;
    }

    while ((!procesos.empty() || !en_procesamiento.empty()))
    {   
        int hptaQueEsEsaMOndaALobien =k - en_procesamiento.size();
        for (int i = 0; i < k - en_procesamiento.size(); i++)
        {
            if (!procesos.empty())
            {
                en_procesamiento.push_back(procesos.front());
                procesos.pop_front();
            }
        }

        int value = *min_element(en_procesamiento.begin(), en_procesamiento.end());

        vector<int> auxiliar;
        for (auto valor : en_procesamiento)
        {
            if (valor - value > 0)
            {
                auxiliar.push_back(valor - value);
            }
        }

        en_procesamiento = auxiliar;


        maximo = max(maximo, value);
    }

    cout << maximo << endl;

    return 0;
}
